index.bs

```diff
@@ -292,8 +338,8 @@ spec: HEADER-STRUCTURE; urlPrefix: https://httpwg.org/http-extensions/draft-ietf
 
       1. Otherwise, [=set/for each=] |item| in the <a>allowlist</a>:
 
-        1. If |item| is [=same origin-domain=] with <var>origin</var>, then
-           return true.
+        1. If the result of running <a>Does url match expression in origin with redirect count?</a>
+           on <var>origin</var>, |item|, <var>origin</var>, and 0 is true then return true.
```

Comment:
The 0 here is irrelevant, right? It looks like that only comes into effect if there is a path component, which we won't pass in?

Reply:
Also, I'm not sure that we can pass `origin` in as the first parameter; the algorithm takes a URL, which an origin is not.

The second origin is interesting -- it should actually be the origin of the policy, which isn't part of the allowlist, since the algorithm handles items like 'self' here. However, we explicitly expand 'self' below when constructing allowlists. I wonder if this should be some null value instead, so that it cannot affect the match in any way?

Reply:
We need to pass in the 0 since the algo asks for it though right?

Should I convert the origin to a string and then parse it as a URL? I assumed we could pass it here since it has a subset of the parts.

In the chrome source we are just passing the origin of the page twice https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/common/permissions_policy/origin_with_possible_wildcards.cc;drc=59515e08807151f06afcacfb2063732dd403ea05;l=116. I think that's the correct thing to do in this instance since we're doing all checks relative to the origin of the prospective page.

Reply:
> We need to pass in the 0 since the algo asks for it though right?

Yeah, I just wanted to make sure that it didn't actually matter -- you could pass in 7, or 100000000, and it wouldn't make a difference since there should be no path component.

> Should I convert the origin to a string and then parse it as a URL? I assumed we could pass it here since it has a subset of the parts.

`origin` can be opaque, in which case there would be no valid serialization, and we actually do need to see if it is same-origin with `item`. At least, that's the case currently. I think there may be no way for an allowlist to contain an opaque origin with this change.

That might be a more subtle change in the spec than I thought at first: Initially an allowlist would only contain concrete origins which were all resolved during parsing (unless the whole allowlist was `*`), but now it contains a list of patterns to match against, which cannot be made concrete.

> In the chrome source we are just passing the origin of the page twice https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/common/permissions_policy/origin_with_possible_wildcards.cc;drc=59515e08807151f06afcacfb2063732dd403ea05;l=116. I think that's the correct thing to do in this instance since we're doing all checks relative to the origin of the prospective page.

That algorithm, as I read it, takes \<absolute url of resource to check\>, \<pattern to check against\>, \<origin to resolve pattern against\>, \<redirect count\>. Because \<pattern\> could be `self`, which needs to resolve to the containing page's URL, that second origin should be the policy-owning-page's origin.

I think it doesn't matter right now, because we pre-resolve items like `self` during parsing, and don't store them in the allowlist, but if we need to change that, then the second origin will need to be that of the page with the policy.

Reply:
Would it be enough to add a check that the origin wasn't opaque before calling this then?